### PR TITLE
add types

### PR DESCRIPTION
**Pull request by tjenkinson**
This adds types for TypeScript.

image

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -0,0 +1,6 @@
+declare function sanitize(
+  input: string,
+  options?: { replacement?: string }
```
Comment (joelmukuthu) [inline index.d.ts]: `replacement` can also be a function ([ref](https://github.com/parshap/node-sanitize-filename/blob/master/README.md#sanitizeinputstring-options))

**Comment by joelmukuthu** [comment]
@parshap are you okay with having types in here rather than the definitions on [DefinitelyTyped](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/master/types/sanitize-filename)? IMO they’re are sufficiently small that they shouldn’t cause a maintenance burden.

File: index.js [hunk 2]
```diff
@@ -24,7 +24,7 @@
  * http://unix.stackexchange.com/questions/32795/what-is-the-maximum-allowed-filename-and-folder-size-with-ecryptfs
  *
  * @param  {String} input   Original filename
- * @param  {Object} options {replacement: String}
+ * @param  {Object} options {replacement: String | Function }
```
Comment (parshap) [inline index.js]: Thanks for this correction!

**Comment by parshap** [comment]
Published in `sanitize-filename@1.6.3`.

**Comment by joelmukuthu** [comment]
I created a [PR over at DefinitelyTyped](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/37917) to remove those definitions.

**Comment by theFroh** [comment]
Was that TS test done with `"esModuleInterop": true`?

**Comment by theFroh** [comment]
Ah, righto. I think that's just a change in the Definitely Typed definitions versus these new bundled ones, then.

**Pull request merged**